### PR TITLE
Add more infor about function pointers to TRPL

### LINKED ISSUE #26746: Function pointers don't seem to be mentioned in the book
The fact that function pointers implement the closure traits `Fn`/`FnMut`/`FnOnce` also doesn't seem to be mentioned in either the book or the reference.

### PR DESCRIPTION
**Pull request by steveklabnik**
1. mention them in the function chapter
2. mention their coercion to closures in the closures chapter

Fixes #26746

### CODE REVIEW

File: src/doc/trpl/closures.md [hunk 1]
```diff
@@ -316,6 +316,35 @@ assert_eq!(3, answer);
...
+
+let answer = call_with_one(&f);
+
+assert_eq!(2, answer);
+```
+
+In this example, we don’t strictly need the intermediate variable `f`,
+the name of the function works just fine too:
+
+```ignore
```
Comment (killercup) [inline src/doc/trpl/closures.md]: Should be `rust,ignore`

Comment (steveklabnik) [inline src/doc/trpl/closures.md]: `rust` is assumed by default

Comment (killercup) [inline src/doc/trpl/closures.md]: @steveklabnik Only when using rustdoc, but not when generating the ebooks. I'd like to keep the Markdown sources as universal as possible. Don't worry about it, though. I'll send a PR later today to add `rust` where necessary.

**Pull request merged**